src/uu/cksum/src/cksum.rs

```diff
@@ -141,6 +174,11 @@ where
         let (sum, sz) = digest_read(&mut options.digest, &mut file, options.output_bits)
             .map_err_context(|| "failed to read input".to_string())?;
 
+        if options.raw {
+            let bytes_str = sum.parse::<u32>().unwrap().to_be_bytes();
```

Comment:
Depending on the specified algorithm, this `unwrap()` fails :|  For example:
```
$ cargo run cksum LICENSE --raw -a sm3
thread 'main' panicked at /home/dho/projects/coreutils/src/uu/cksum/src/cksum.rs:178:48:
called `Result::unwrap()` on an `Err` value: ParseIntError { kind: InvalidDigit }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
It works fine for `crc`, `sysv`, and `bsd`, but not the other algorithms.

Reply:
Thanks! I will add a test for raw with each algorithm and try to fix that.